crossbar/router/test/test_router.py

```diff
@@ -32,13 +32,42 @@
 
 from twisted.trial import unittest
 
+from mock import Mock
+
 import txaio
 
+import six
+
 from autobahn.wamp import types
 from autobahn.twisted.wamp import ApplicationSession
 
-from crossbar.router.router import RouterFactory
+from crossbar.router.router import RouterFactory, CrossbarRouterFactory
 from crossbar.router.session import RouterSessionFactory
+from crossbar.router.session import CrossbarRouterSession
+
+
+class TestCrossbarSessions(unittest.TestCase):
+    def test_onjoin_metaevent(self):
+        # we're just short-circuiting straight to calling onJoin()
+        # ourselves rather than drive the protocol to that point.
+        factory = CrossbarRouterFactory()
+        session = CrossbarRouterSession(factory)
+        # ...but we therefore have to set up enough internals so the
+        # onJoin() can emit the session-details it wants.
+        session._router = Mock()
+        session._router._realm = Mock()
+        session._router._realm.session = Mock()
+        session._transport = Mock()
+        session._transport._transport_info = 'nothing to see here'
+        details = types.SessionDetails(u'test_realm', 1234)
+
+        session.onJoin(details)
+
+        calls = session._router._realm.session.method_calls
+        print(calls, calls[0][1])
```

Comment:
Ah, nice catch. See also a different (more robust) fix, in autobahn under [PR 390](https://github.com/tavendo/AutobahnPython/pull/390).